src/sql/platform/connection/common/connectionManagementService.ts

```diff
@@ -562,15 +573,20 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 			let serverInfo: azdata.ServerInfo = this.getServerInfo(connectionProfile.id);
 			let profile: interfaces.IConnectionProfile = connectionProfile.toIConnectionProfile();
 			iconProvider.getConnectionIconId(profile, serverInfo).then(iconId => {
-				if (iconId) {
-					this._connectionIconIdCache.set(connectionProfile.id, iconId);
+				if (iconId && this._mementoObj && this._mementoContext) {
+					if (!this._mementoObj.CONNECTION_ICON_ID) {
+						this._mementoObj.CONNECTION_ICON_ID = <any>{};
+					}
+					this._mementoObj.CONNECTION_ICON_ID[connectionProfile.id] = iconId;
+					this._mementoContext.saveMemento();
 				}
 			});
 		}
 	}
 
 	public getConnectionIconId(connectionId: string): string {
-		return this._connectionIconIdCache.get(connectionId);
+		if (!connectionId || !this._mementoObj || !this._mementoObj.CONNECTION_ICON_ID) { return undefined; }
```

Comment:
nit: body of it statements on separate lines please (even for single lines). Easy to misread this as executing the below line instead

Reply:
fixed.